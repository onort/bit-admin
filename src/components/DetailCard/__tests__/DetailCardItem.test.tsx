import React from "react"
import { shallow } from "enzyme"

import { DetailCardItem } from "../"

const testCardItemTitle = "TestCardItemTitle"
const testCardItemContent = "TestCardItemContent"
const testClassName = "testClassName"
const testUrl = "http://example.com"

describe("<DetailCardItem />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <DetailCardItem
        content={testCardItemContent}
        linked={true}
        url={testUrl}
        title={testCardItemTitle}
      />
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should display passed content and title", () => {
    const wrapper = shallow(
      <DetailCardItem content={testCardItemContent} title={testCardItemTitle} />
    )
    expect(wrapper.find(".itemTitle").text()).toBe(testCardItemTitle)
    expect(wrapper.find(".itemContent").text()).toBe(testCardItemContent)
  })

  it("should display have a link if it's linked and passed a url", () => {
    const wrapper = shallow(
      <DetailCardItem
        content={testCardItemContent}
        linked={true}
        url={testUrl}
        title={testCardItemTitle}
      />
    )
    expect(wrapper.find("a").exists()).toBe(true)
    expect(wrapper.find("a").prop("href")).toBe(testUrl)
  })

  it("should be able to have custom className", () => {
    const wrapper = shallow(
      <DetailCardItem
        className={testClassName}
        content={testCardItemContent}
        title={testCardItemTitle}
      />
    )
    expect(wrapper.find(`.${testClassName}`).exists()).toBe(true)
  })
})
