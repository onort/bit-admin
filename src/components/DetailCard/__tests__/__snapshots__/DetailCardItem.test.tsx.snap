// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailCardItem /> should match snapshot 1`] = `
<div
  className="itemContainer"
>
  <span
    className="itemTitle"
  >
    TestCardItemTitle
  </span>
  <span
    className="itemContent"
  >
    <a
      className="itemLink"
      href="http://example.com"
      target="_blank"
    >
      TestCardItemContent
      <MdOpenInNew />
    </a>
  </span>
</div>
`;

exports[`<DetailCardItem /> should match snapshot 2`] = `"<div class=\\"itemContainer\\"><span class=\\"itemTitle\\">TestCardItemTitle</span><span class=\\"itemContent\\"><a class=\\"itemLink\\" href=\\"http://example.com\\" target=\\"_blank\\">TestCardItemContent<svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z\\"></path></svg></a></span></div>"`;
